### PR TITLE
wireguard: enable wireguard kernel module

### PR DESCRIPTION
**Pull request by trungams**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

Customer expects wireguard to be shipped as part of kernel modules for their VPN solution. This change enables wireguard as a loadable kernel module.

Link to Kconfig file that contains wireguard configs: https://github.com/torvalds/linux/blob/master/drivers/net/Kconfig#L75

###### Change Log  

- Update kernel config files for x86_64 and aarch64
- Update config file hashes
- Bump release number

###### Does this affect the toolchain?  

**YES**

###### Associated issues  

- https://microsoft.visualstudio.com/OS/_workitems/edit/42861525/

###### Test Methodology

- Local build succeeded
- Deployed VM and verified wireguard is loadable and can be used by other tools (ip)
- Buddy build:
  - aarch64 [299722](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=299722)
  - x84_64 [299723](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=299723)

### CODE REVIEW

File: SPECS/kernel/kernel.spec [hunk 1]
```diff
@@ -203,7 +200,7 @@ for MODULE in `find %{buildroot}/lib/modules/%{uname_r} -name *.ko` ; do \
 %define __spec_install_post\
     %{?__debug_package:%{__debug_install_post}}\
     %{__arch_install_post}\
-    %{__os_install_post}\
+    %__os_install_post\
```
Comment (rlmenge) [inline SPECS/kernel/kernel.spec:203]: I am not confident this change is desired based on the changelog. I assume this is something the linter suggested. Do you know if this behaves as expected?

File: SPECS/kernel/config_aarch64 [hunk 2]
```diff
@@ -2981,7 +2981,8 @@ CONFIG_MII=y
 CONFIG_NET_CORE=y
 CONFIG_BONDING=m
 CONFIG_DUMMY=m
-# CONFIG_WIREGUARD is not set
+CONFIG_WIREGUARD=m
```
Comment (rlmenge) [inline SPECS/kernel/config_aarch64:2984]: Can you add a link to the kconfig for this config in the description?

Comment (trungams) [reply SPECS/kernel/config_aarch64:2984]: I added a link to kconfig

File: SPECS/kernel/kernel.spec (hunk 1 above)
Comment (trungams) [reply SPECS/kernel/kernel.spec]: Yeah. It looks like build pipelines failed soon after. I reverted the change and re-run the pipelines for both amd64 and arm64. Both succeeded. Ignoring linter output for now.

**Review by rmhsawyer — APPROVED** [review]

**Review by rlmenge — APPROVED** [review]

**Pull request merged**